ros/src/tl_detector/tl_detector.py

```diff
@@ -293,33 +338,25 @@ def process_traffic_lights(self):
 
         """
 
-        # searching_distance_tl parameter sets distance in which
+        # searching_dist_tl parameter sets distance in which
         # traffic lights will be searched as number of waypoints
-        searching_distance_tl = 120
-
+        searching_dist_tl = 200
+        light_wp_idx = -1
         light = None
-        light_wp = -1
         if(self.pose and self.waypoints):
-            # In case that array with traffic light waypoints does not exist,
-            # create it
-            if(self.tl_waypoints == []):
-                self.get_tl_waypoints()
-            car_position_wp = self.get_closest_waypoint_idx(self.pose.pose)
-            # Loop thorugh waypoints to find the closest traffic ligt waypoint
-            smallest_tl_distance = 10000
-            for tl_waypoint in self.tl_waypoints:
-                distance_between_waypoints = abs(car_position_wp-tl_waypoint)
-                if(distance_between_waypoints < searching_distance_tl and
-                    distance_between_waypoints < smallest_tl_distance and
-                        car_position_wp < tl_waypoint):
-                    light_wp = tl_waypoint
-                    smallest_tl_distance = distance_between_waypoints
+            car_pos_wp_idx = self.get_closest_waypoint_idx(self.pose.pose)
+            light_wp_idx = self.get_closest_tl_wp_idx(
+                car_pos_wp_idx, searching_dist_tl
+                )
 
         # If waypoint has been found get traffic light state
-        if light_wp > -1:
-            light = self.waypoints.waypoints[light_wp]
+        if light_wp_idx > -1:
+            light = self.waypoints.waypoints[light_wp_idx]
             state = self.get_light_state(light)
-            return light_wp, state
+            print(light_wp_idx)
```

Comment:
Could we `rospy.loginfo()` here?

Reply:
@idanov sure, or I can remove this line completely, I've forgot to remove it after testing.